### PR TITLE
end_round(): Pass turn back to the original leader when there is no surpass.

### PR DESCRIPTION
**Pull request by fallspectrum**
I noticed that when only copies/pivots occur in a turn, and end_round is executed, turn order needed to be manually passed from the end player, back to the one who still holds the initiative.  This assists with that turn change.  Additionally while looking at the rest of the function, I noticed some unused + redundant logic.

This PR:

- Adds a gitignore for .tts folder
- Sets the starting player turn order based on whomever was assigned first initiative.  (Note that the game still possibly announces the first turn to a different color, but now quickly reassigns it to the expected player.  I've yet to track down why an incorrect player is first assigned when Turns.enable = true is applied.)
- Adjusts Control start_chapter() to broadcast who the starting player is with a newline at the end.
- Adjusts InitiativeMarker to setup a global variable for tracking who the initiative_player is. 
- Refines Control end_round() so that initiative is passed between players more consistently and cleans up the game state broadcasts for readability.

### CODE REVIEW

**Comment by Quinnsicle** [comment]
I think it might be beneficial to broadcast what happens to the initiative.

File: src/Control.lua [hunk 1]
```diff
@@ -136,34 +136,25 @@ function end_round()
         Turns.turn_color = Initiative.player
```
Comment (Quinnsicle) [inline src/Control.lua:136]: Per my comment, you can add...
```
broadcastToAll(Initiative.player .. " seized and gains initiative.")
returrn
```

File: src/Control.lua [hunk 2]
```diff
@@ -136,34 +136,25 @@ function end_round()
...
-                                 tostring(surpassing.number)
-
-        local all_players = Global.getVar("active_players")
-        for _, p in ipairs(all_players) do
-            if (p.last_action_card) then
-                if (p.last_action_card.type == surpassing.type and
-                    p.last_action_card.number == surpassing.number) then
-
+        if not surpassing then
+            Turns.turn_color = Initiative.player
```
Comment (Quinnsicle) [inline src/Control.lua:140]: Add
```
broadcastToAll(Initiative.player .. " keeps initiative.")
returrn
```

File: src/Control.lua [hunk 3]
```diff
@@ -136,34 +136,25 @@ function end_round()
...
-                    return
+                    break
                 end
             end
         end
     end
 
-    broadcastToAll("End Hand\n")
-    return
+    broadcastToAll("End Round\n")
```
Comment (Quinnsicle) [inline src/Control.lua:157]: Change this to
```
broadcastToAll(Initiative.player .. " gains initiative.")
```

**Comment by fallspectrum** [comment]
@Quinnsicle agreed on announcing the game state better.  While adding these broadcast messages I ran into a couple issues.  Broadcast messages from the InitiativeMarker functions themselves created duplicate/noisy message in chat, so in an effort to handle both cases where a player manual clicks "take initiative" on the control board, or if they simply hit end round, a "silent" bool parameter was added to InitiativeMarker.seize and .take.  

Next, the InitiateMarker.player var was locally defined, and all separate bundle includes of InitiativeMarker makes the maintenance of that var fairly inconsistent.  I opt'd to instead have it setUp a global variable, and then call that for the control options.

Lastly, I decided to keep the "End Round" message to make it clear that the initiative changes/seizes are done for that round, coloring it purple so it stands out compared to the normal player colors.

File: src/Control.lua (hunk 3 above)
Comment (fallspectrum) [reply src/Control.lua]: See lower comment

File: src/Control.lua (hunk 2 above)
Comment (fallspectrum) [reply src/Control.lua]: See lower comment

File: src/Control.lua (hunk 1 above)
Comment (fallspectrum) [reply src/Control.lua]: See lower comment

**Comment by fallspectrum** [comment]
I also noticed that the very first game turn is not properly assigned to whomever holds initiative.  Updated start_chapter() to fix that.

I tried applying the turn during base game setup, but for some unknown reason it's not applying (either before or after here: https://github.com/Quinnsicle/arcs_tts/blob/main/src/BaseGame.lua#L242)

**Comment by Quinnsicle** [comment]
The broadcast changes look good. Could you explain the maintenance cost and inconsistency in having the player locally defined in the InitiativeMarker module? I understood modules in Lua to be handled similar to a singleton by default. When a module is "required" and loaded, it's put in a meta table cache. Subsequent requires simply use the cached version of the module instead of instantiating another. Let me know if I'm misunderstanding anything, and if I can help with the early returns.

**Comment by fallspectrum** [comment]
In my opinion the early returns make the chat output less readable/obvious, for example:
end_round_with_returns

Whereas if we remove the early returns, and let "End Round" purple broadcast + newline clearly call the end of the round:
end_round_without_returns

It may be a preference of aesthetics, but let me know your thoughts.

Also, until seize detection is added, I removed the broadcast from the end_round() seize block, as it's a bit redundant in chat once I started seeing/testing it.  Once I finish adding seize detection we can have the broadcast either handled when a manual seize is trigger via the control button OR if a fresh seize is detected during end_round(), rather than firing in both cases.


Regarding needing the global variable, the setup bundle instantiates it's own require InitiativeMarker, so when the BaseGame.setup does the first initiative assignment (here: https://github.com/Quinnsicle/arcs_tts/blob/main/src/BaseGame.lua#L223)

When control end_round() is triggered, it bombs out looking up a nil value on Initiative.player since the control bundle has it's own instantiation/require of InitiativeMarker.  Hence the need to declare a global var.

If there is an alternate or better way to pass variable state between different object bundles let me know.  

I'm adding another change to this PR, setting the player turn before Turns.enable = true is applied in BaseGame + Campain.  I was trying to address this earlier and it wasn't working for some reason, user error.  I was able to get near expected behavior, where turn order is set before start chapter occurs.

**Comment by fallspectrum** [comment]
Updated the PR description to summarize all the changes in the PR, as the scope of the changes grew from the initial submission.

**Comment by Quinnsicle** [comment]
I just wanted the early returns to make the code more readable. I think we could still get the desired functionality with early returns by simply duplicating the "End Round" broadcast before every return. It's not 100% ideal but I don't imagine changing that "End Round" broadcast. Anyways, I don't want to hold up these changes any longer than I have so I'll approve it as is.

**Comment by fallspectrum** [comment]
Superseded by #31

**Pull request closed without merging**